src/develop/masks/gradient.c

```diff
@@ -515,8 +531,10 @@ static int dt_gradient_get_points(dt_develop_t *dev, float x, float y, float rot
   *points = NULL;
   *points_count = 0;
 
+  dt_pthread_mutex_lock(&dev->preview_pipe_mutex);
```

Comment:
yeah all that input width access with a mutex seems like overkill to me. it's really just the size of the according  MIPF (see develop.c:222 dt_dev_pixelpipe_set_input) and only changes when changing images via exit/enter dr mode or spacebar/filmstrip.

about these widths/heights (backbuf same as input) i'm not all that concerned. worst thing to happen is that the gui jumps weirdly if it picks up a new backbuf width before the height is written and will be refreshed a millisecond after.

but the backbuf mutex can be locked in a reasonably fine-grained manner, so i think we should do it.

input only matters during image switching, and the preview_pipe_mutex is a pretty big thing, so not sure if worth it.